content/shared/v3-core-plugins/_index.md

```diff
@@ -0,0 +1,297 @@
+
+## Schedule Plugin
+Schedule plugins run on a schedule specified in cron syntax. The plugin will receive the local API, the time of the trigger, and any arguments passed in the trigger definition. Here's an example of a simple schedule plugin:
+
+```python
+# see if a table has been written to in the last 5 minutes
+def process_scheduled_call(influxdb3_local, time, args=None):
+    if args and "table_name" in args:
+        table_name = args["table_name"]
+        result = influxdb3_local.query(f"SELECT * FROM {table_name} WHERE time > now() - 'interval 5m'")
+        # write an error log if the result is empty
+        if not result:
+            influxdb3_local.error(f"No data in {table_name} in the last 5 minutes")
+    else:
+        influxdb3_local.error("No table_name provided for schedule plugin")
+```
+
+### Schedule Trigger Configuration
+Schedule plugins are set with a `trigger-spec` of `schedule:<cron_expression>`. The `args` parameter can be used to pass configuration to the plugin. For example, if we wanted the above plugin to run the check every minute, we would use `schedule:*/5 * * * *` as the `trigger-spec`.
+
+## On Request Plugin
+On Request plugins are triggered by a request to a specific endpoint under `/api/v3/engine`. The plugin will receive the local API, query parameters `Dict[str, str]`, request headers `Dict[str, str]`, request body (as bytes), and any arguments passed in the trigger definition. Here's an example of a simple On Request plugin:
+
+```python
+import json
+
+def process_request(influxdb3_local, query_parameters, request_headers, request_body, args=None):
+    for k, v in query_parameters.items():
+        influxdb3_local.info(f"query_parameters: {k}={v}")
+    for k, v in request_headers.items():
+        influxdb3_local.info(f"request_headers: {k}={v}")
+
+    request_data = json.loads(request_body)
+
+    influxdb3_local.info("parsed JSON request body:", request_data)
+
+    # write the data to the database
+    line = LineBuilder("request_data").tag("tag1", "tag1_value").int64_field("field1", 1)
+    # get a string of the line to return as the body
+    line_str = line.build()
+
+    influxdb3_local.write(line)
+
+    return 200, {"Content-Type": "application/json"}, json.dumps({"status": "ok", "line": line_str})
+```
+
+### On Request Trigger Configuration
+On Request plugins are set with a `trigger-spec` of `request:<endpoint>`. The `args` parameter can be used to pass configuration to the plugin. For example, if we wanted the above plugin to run on the endpoint `/api/v3/engine/my_plugin`, we would use `request:my_plugin` as the `trigger-spec`.
+
+Trigger specs must be unique across all configured plugins, regardless of which database they are tied to, given the path is the same.
```

Comment:
```suggestion
Because all On Request triggers for a server use the same `/api/v3/engine` root path, trigger specs must be unique across all configured plugins, regardless of which database they are associated with.
```